source/numerics.tex

```diff
@@ -7279,40 +7250,42 @@
 \indexlibrarymember{operator\shl=}{valarray}%
 \indexlibrarymember{operator\shr=}{valarray}%
 \begin{itemdecl}
-valarray& operator*= (const valarray&);
-valarray& operator/= (const valarray&);
-valarray& operator%= (const valarray&);
-valarray& operator+= (const valarray&);
-valarray& operator-= (const valarray&);
-valarray& operator^= (const valarray&);
-valarray& operator&= (const valarray&);
-valarray& operator|= (const valarray&);
-valarray& operator<<=(const valarray&);
-valarray& operator>>=(const valarray&);
+valarray& operator*= (const valarray& v);
+valarray& operator/= (const valarray& v);
+valarray& operator%= (const valarray& v);
+valarray& operator+= (const valarray& v);
+valarray& operator-= (const valarray& v);
+valarray& operator^= (const valarray& v);
+valarray& operator&= (const valarray& v);
+valarray& operator|= (const valarray& v);
+valarray& operator<<=(const valarray& v);
+valarray& operator>>=(const valarray& v);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
+\requires
+\tcode{size() == v.size()}.
 Each of these operators may only be instantiated for a type \tcode{T}
-to which the indicated operator can be applied.
-Each of these operators
-performs the indicated operation on each of its elements and the
-corresponding element of the argument array.
+to which the indicated operator can be applied. 
```

Comment:
Do we want to be more precise in these requirements? Being able to apply the operator doesn't mean much if we don't say what types the operands are. A type could overload `T::operator*=(std::string)` but that doesn't mean we can use `valarray<T>{} *= T{}`.

Maybe something like "For each operator function, the operator that it implements is invocable with two operands of type `T`"